### ISSUE TITLE
[Book] Add reference to models repo

### ISSUE THREAD

**Issue by laggui** [tracel-ai/burn#1808]
We're starting to have more and more models available for our users so it would be nice to add a small paragraph in the book with a link to the models repo.

**Comment by polarathene** [comment]
Probably worth a mention on the website too for those who land there?

I noticed the benchmarks (in this repo, and the separate benchmarks repo) lack any contributed results. It'd be nice if someone could provide one for users to get an impression upfront?

**Comment by laggui** [comment]
Good idea! We should probably add the models to the README as well, following the examples (which should probably be better enumerated).
